sandbox/modules/identity/src/main/java/org/opensearch/identity/rest/action/permission/check/CheckPermissionRequest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.rest.action.permission.check;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.ToXContent;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+
+import static org.opensearch.action.ValidateActions.addValidationError;
+
+public class CheckPermissionRequest extends ActionRequest implements ToXContentObject {
+
+    private String principalString;
```

Comment:
For future PR: We've got those principal identifier tokens, could we accept either a PIT or the principal string via the same request object?

Reply:
That is the hope. I was thinking that using the tokens and calling their toString() would let us pass them as the principal/user and then we can use these same calls to see what type of permissions any plugins or extensions have.